File: 프로그래머스/2/42587. 프로세스/프로세스.java
import java.util.*;

class Solution {
    public int solution(int[] priorities, int location) {
        
        Deque<int[]> que = new ArrayDeque<>();
        PriorityQueue<Integer> pq = new PriorityQueue<>(Collections.reverseOrder());
        
        for(int i=0; i<priorities.length; i++){
            int num = priorities[i];
            que.add(new int[]{num, i});
            pq.add(num);
        }
        
        int answer = 0;
        int cnt = 0;
        
        while(!que.isEmpty()){
            int now[] = que.poll();
            if(pq.peek() == now[0]){
                pq.poll();
                cnt++;
                if(now[1] == location){
                    answer = cnt;
                }
            }else{
                que.add(now);
            }
        }
       
        return answer;
    }
}

File: 프로그래머스/2/42587. 프로세스/프로세스.js
function solution(priorities, location) {
    
    const que = [];
    const pq = [];
    
    for(let i=0; i<priorities.length; i++){
        var num = priorities[i];
        que.push([num, i]);
        pq.push(num);
    }
    
    pq.sort((a,b) => b-a);
    
    var answer = 0;
    var cnt = 0;
    
    while(que.length){
        let now = que.shift();
        if(now[0] == pq[0]){
            pq.shift();
            cnt++;
            if(now[1] == location){
                answer = cnt;
                break;
            }
        }else{
            que.push(now);
        }
    }
    
    return answer;
}